repair/row_level.cc

```diff
@@ -3016,10 +3025,11 @@ class row_level_repair_gossip_helper : public gms::i_endpoint_state_change_subsc
         : _repair_service(repair_service)
     {}
     future<> remove_row_level_repair(gms::inet_address node) {
-        rlogger.debug("Started to remove row level repair on all shards for node {}", node);
+        const auto id = _repair_service.get_gossiper().get_host_id(node);
```

Comment:
Given the purpose of the commit message is not to depend on gossiper propagation to be quick enough -- doesn't this line kinda defeat the purpose?

Reply:
So the scenario was:
- we replace-with-same-ip, old ID = X new ID = Y, IP = .1 let's say
- we start repair and create repair_meta for .1
- now for some reason we get gossiper notification with ID = X IP = .1
- we remove repair meta for IP .1 which is wrong

And after your changes we have:
- start repair and create repair_meta for ID = Y
- for some reason we get gossiper notification with ID = X IP = .1
- we remove repair meta for ID = X i.e. we don't remove anything because there is no repair meta

Correct?

Reply:
So in this scenario `get_host_id(.1)` will return `X` (old ID)

Reply:
It seems we no longer need this commit, `wait_for_ip` ensures roughly the same invariants as they were before this PR. I dropped this commit for now.